x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/action/ResponseValueUtils.java

```diff
@@ -84,6 +85,30 @@ public static Iterator<Iterator<Object>> pagesToValues(List<String> dataTypes, L
         );
     }
 
+    /**
+     * Returns an iterator of values for the given column.
+     */
+    static Iterator<Object> valuesForColumn(int columnIndex, String dataType, List<Page> pages) {
+        BytesRef scratch = new BytesRef();
+        return Iterators.flatMap(pages.iterator(), page -> Iterators.forRange(0, page.getPositionCount(), pos -> {
```

Comment:
There needs to be a check for the underlying page, if it was check during the iteration.

Reply:
If this is related to access-after-close, then that already is covered. There is a page closure check when the iterator implementation retrieves the block from the page. Does that address the concern, or have I missed the point?

Reply:
I was expecting some kind of check from the super class (checkOpen() or if (closed==true) throws). Is there a test for this,?I might have missed it.

Reply:
I beefed up the closed state checking and added some more test scenarios.